sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3416,6 +3416,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val DECORRELATE_EXISTS_AND_IN_SUBQUERIES =
+    buildConf("spark.sql.optimizer.decorrelateExistsIn.enabled")
```

Comment:
Does this flag depend on this `DECORRELATE_INNER_QUERY_ENABLED` flag?

Reply:
no, those are independent. The DECORRELATE_INNER_QUERY_ENABLED flag is for scalar/lateral subqueries, and the current one is for IN/EXISTS.

Reply:
Looks like for IN/EXISTS to be decorrelated you need both flags enabled, which makes sense.

Reply:
ack!